recruiter/views.py

```diff
@@ -27,14 +28,32 @@ def recruiter_view_my_jobs_and_applications(request):
     return render(request, "recruiter_my_jobs_and_applications.html", context)
 
 
-class UpdateRecruiterSettings(UserPassesTestMixin, UpdateView):
-    model = Recruiter
-    fields = ('name', 'company', 'email', 'phone_number')
-    template_name = 'recruiter_account_settings.html'
-    success_url = '/'
+def account_update_success(request):
+    context = {}
+    add_is_recruiter_to_context(request, context)
+    return render(request, 'account_settings_update_success.html', context)
 
-    def test_func(self):
-        return self.request.user.id == int(self.kwargs['pk'])
+
+def update_recruiter_account_settings_view(request):
+    context = {}
+    jobboard.views.add_navbar_links_to_context(request, context)
+    recruiter_object = get_object_or_404(Recruiter, user_id=request.user.id)
+    update_recruiter_form = UpdateRecruiterAccountSettingsForm(request.POST or None, instance=recruiter_object)
+    add_is_recruiter_to_context(request, context)
+    context["form"] = update_recruiter_form
+    if request.method == 'GET':
+        return render(request, "recruiter_account_settings.html", context)
+
+    elif request.method == "POST" and update_recruiter_form.is_valid():
+        update_recruiter_form.save()
+        return redirect("/account_update_success/", context)
+
+    elif request.method == "POST" and not update_recruiter_form.is_valid():
+        return render(request, "recruiter_account_settings.html", context)
+
+
+def add_is_recruiter_to_context(request, context):
```

Comment:
I think you can delete this function as the "is_recruiter" key is no longer required in the context dict.

Reply:
fixed